src/analyses/call_graph_helpers.cpp

```diff
@@ -101,3 +101,25 @@ std::set<irep_idt> get_functions_reachable_within_n_steps(
   return get_functions_reachable_within_n_steps(
     graph, start_functions, n, forwards);
 }
+
+std::set<irep_idt> get_shortest_function_path(
+  const call_grapht::directed_grapht &graph,
+  const irep_idt &src,
+  const irep_idt &dest)
+{
+  std::set<irep_idt> result;
+  std::size_t start_node = *graph.get_node_index(src);
+  std::size_t dest_node = *graph.get_node_index(dest);
+
+  PRECONDITION(
```

Comment:
As the other instance, dereferencing an Empty `optionalt` throws an exception, so make sure you check for emptiness (usually with `.has_value()` or the implicit boolean conversion (e.g. `if(graph.get_node_index(src)) { // now safe to use *graph.get_node_index(src) }`) rather than comparison with `optionalt<std::size_t>()` before using `operator*`